v3/src/js/views/venues/VenueDetailRow.jsx

```
@@ -0,0 +1,71 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { flatten, noop } from 'lodash';
+import { arrangeLessonsForWeek } from 'utils/timetables';
+import { colorLessonsByKey } from 'utils/colors';
+import Timetable from 'views/timetable/Timetable';
+
+import type { DayAvailability } from 'types/venues';
+import type { Venue } from 'types/modules';
+
+import styles from './VenueDetailRow.scss';
+
+type Props = {
+  name: Venue,
+  availability: DayAvailability[],
+  expanded: boolean,
+  onClick: (Venue, string) => void,
+  rootElementRef?: (HTMLElement) => void, // For parent components to obtain a ref to the root HTMLElement
+}
+
+export default class VenueDetailRow extends PureComponent<Props> {
+  static defaultProps = {
+    name: '',
+    availability: [],
+    expanded: false,
+    onClick: noop,
+  };
+
+  arrangedLessons() {
+    if (!this.props.expanded) {
+      return null;
+    }
+
+    const availability: DayAvailability[] = this.props.availability;
+    // const lessons = flatMap(availability, a => a.Classes) // Not using flatMap as it results in a Flow error
```

Remove this if not using.

---

Actually, if this is a libdef issue, it may be better to file issue/PR against https://github.com/flowtype/flow-typed, and just use a `$FlowFIxMe` to ignore the error while the issue/PR is getting processed.

---

I'm not familiar enough with Flow to know if it's an issue with libdef though. I'll just change it to `$FlowFixMe` first and figure this out later